ADR/0017-posture-keeper.md

```diff
@@ -0,0 +1,59 @@
+# 17. Posture Keeper
+
+Date: 2023-02-18
+
+## Status
+
+Accepted
+
+## Context
+Posture keeper is a REST-based web service that allows clients to store and retrieve SAST analysis data for a specific git repository and commit SHA. The service is designed to be scalable, reliable, and secure, and uses MongoDB as its underlying data store. The service is implemented using Quarkus. By utilizing this service StoneSoup is going to be able to provides experiences that reflects the current state of security and/or quality posture of the applications.
+
+## Design Goals
+
+The following are the key design goals of the service:
+
+* Support storing and retrieving SAST analysis data for a specific git repository and commit SHA
+* Accept and produce JSON-formatted data
+* Use MongoDB as the underlying data store for efficient data retrieval and storage
+* Support secure authentication and authorization to ensure only authorized clients can upload and retrieve data
+* Implement rate limiting for the POST endpoint to prevent abuse and ensure optimal performance
+* Implement pagination for the GET endpoint to efficiently retrieve large volumes of data
+
+## Architecture
+
+The Posture Keeper is implemented as a microservice using Quarkus. The architecture consists of the following components:
+
+### Client
+
+The client is responsible for making requests to the service endpoints using HTTP. Clients can be web applications, mobile applications, or any other type of application that can communicate over HTTP.
+
+### REST API
+
+The REST API is implemented using Quarkus. The API provides endpoints for storing and retrieving SAST analysis data for a specific git repository and commit SHA. It also provides an endpoint for comparing two analysis results and returning the new defects.
```

Comment:
What would be the basis of an analysis result? Would we be on a snapshot level or a component level?

Reply:
FWIW, on reading this I had assumed it would be on a component level - due to the retrieval parameters being "git repository and commit SHA".